ethereumj-core/src/main/java/org/ethereum/sharding/processing/consensus/BeaconStateTransition.java

```diff
@@ -67,8 +84,19 @@ public BeaconState applyBlock(Beacon block, BeaconState to) {
                     .withDynasty(dynasty)
                     .withLastStateRecalc(cycleStartSlot(block))
                     .withFinality(finality);
+
+            // remove attestations older than last_state_recalc
+            List<AttestationRecord> uptodateAttestations = new ArrayList<>();
+            for (AttestationRecord record : activeState.getPendingAttestations()) {
+                if (record.getSlot() >= crystallized.getLastStateRecalc()) {
+                    uptodateAttestations.add(record);
+                }
+            }
+            beaconAttester.removeOldSlots(crystallized.getLastStateRecalc());
```

Comment:
The same "pub/sub" note as previous one. I guess it could be even processed by same handler.